### PR TITLE
Spectrum device functionality

### PR DESCRIPTION
**Pull request by wersal454**
## What type of PR is this?
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement
4. [ ] Miscellaneous

### What have you changed and why?
Information:

Integrated spectrum device functionality script + modified it to fit antistasi

### Please verify the following.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?

1. [x] No, not really but you can check it out though
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:

********************************************************
Notes:

### CODE REVIEW

**Comment by wersal454** [comment]
If it comes to pass, might wanna add UGVs to modsets that have them

**Comment by wersal454** [comment]
image
image
Permission granted by the author (11.05.2024)

https://www.youtube.com/watch?v=VHLAAkV5Kxk&t=18s
comments under authors video*

File: A3A/addons/core/functions/init/fn_initServer.sqf [hunk 1]
```diff
@@ -50,6 +50,8 @@ Info("Background init started");
 
 // No reason not to do this early
 [] execVM QPATHTOFOLDER(Scripts\fn_advancedTowingInit.sqf);
+[] execVM QPATHTOFOLDER(Scripts\SpectumDevice\spectrum_device.sqf);
+[] execVM QPATHTOFOLDER(Scripts\SpectumDevice\sa_ewar.sqf);
```
Comment (Silence-112) [inline A3A/addons/core/functions/init/fn_initServer.sqf]: Does the server really need this too?

File: A3A/addons/scrt/Encounter/fn_encounter_fleeingCivConvoy.sqf
Comment (Silence-112) [inline A3A/addons/scrt/Encounter/fn_encounter_fleeingCivConvoy.sqf:1]: Does this literally just change what was already commented out?

File: A3A/addons/core/functions/init/fn_initClient.sqf [hunk 2]
```diff
@@ -41,6 +41,8 @@ if !(isServer) then {
     call A3A_fnc_initVarCommon;
 
     [] execVM QPATHTOFOLDER(Scripts\fn_advancedTowingInit.sqf);
+    [] execVM QPATHTOFOLDER(Scripts\SpectumDevice\spectrum_device.sqf);
+    [] execVM QPATHTOFOLDER(Scripts\SpectumDevice\sa_ewar.sqf);
```
Comment (Silence-112) [inline A3A/addons/core/functions/init/fn_initClient.sqf]: Add a parameter to disable this and it'll probably be fine
```suggestion
    if (enableSpectrumDevice) then {
        [] execVM QPATHTOFOLDER(Scripts\SpectumDevice\spectrum_device.sqf);
        [] execVM QPATHTOFOLDER(Scripts\SpectumDevice\sa_ewar.sqf);
    };
```

**Review by Silence-112 — CHANGES_REQUESTED** [review]

File: A3A/addons/core/functions/init/fn_initServer.sqf (hunk 1 above)
Comment (wersal454) [reply A3A/addons/core/functions/init/fn_initServer.sqf]: Not sure, I just copied what advanced towing had, so... probably?

File: A3A/addons/scrt/Encounter/fn_encounter_fleeingCivConvoy.sqf
Comment (wersal454) [reply A3A/addons/scrt/Encounter/fn_encounter_fleeingCivConvoy.sqf:1]: Yeah, nvm that

File: A3A/addons/core/functions/init/fn_initClient.sqf (hunk 2 above)
Comment (wersal454) [reply A3A/addons/core/functions/init/fn_initClient.sqf]: If you can, confirm that this check works, if so, I'll just merge it

File: A3A/addons/core/Params.hpp [hunk 3]
```diff
@@ -512,6 +512,14 @@ class Params
         texts[] = {$STR_antistasi_dialogs_generic_button_no_text, $STR_antistasi_dialogs_generic_button_yes_text};
         default = 0;
     };
+    class enableSpectrumDevice
+    {
+        attr[] = {"server"};
```
Comment (Silence-112) [inline A3A/addons/core/Params.hpp]: This has `attr[] = {"server"}`, but you're also checking it in `initClient`. Does it still work?

Comment (wersal454) [reply A3A/addons/core/Params.hpp]: Does work, though haven't tested it on dedicated server

Comment (Silence-112) [reply A3A/addons/core/Params.hpp]: Would probably be better to just remove this if the client is checking it too
```suggestion
    class enableSpectrumDevice
    {
```

**Review by Silence-112 — APPROVED** [review]
Ready for testing

**Pull request merged**